File: Dictionaries and Sets/2206. Divide Array Into Equal Pairs/2206. Divide Array Into Equal Pairs.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var divideArray = function (nums) {
  sett = new Set();

  for (const n of nums) {
    if (sett.has(n)) sett.delete(n);
    else sett.add(n);
  }

  return sett.size === 0;
};

// Time: O(n)
// Space: O(n)

File: Dictionaries and Sets/2206. Divide Array Into Equal Pairs/2206. Divide Array Into Equal Pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        sett = set()
        for n in nums:
            if n in sett:
                sett.remove(n)
            else:
                sett.add(n)

        return len(sett) == 0

# Time: O(n)
# Space: O(n)
